bigbluebutton-web/grails-app/controllers/org/bigbluebutton/web/controllers/ApiController.groovy

```diff
@@ -1231,1076 +1549,602 @@ class ApiController {
-      invalid("checksumError", "You did not pass the checksum security check")
-      return
-    }
-    // END - backward compatibility
+        if (errors.hasErrors()) {
+            respondWithErrors(errors)
+            return
+        }
 
-    ApiErrors errors = new ApiErrors()
+        // Do we agree on the checksum? If not, complain.
+        if (!paramsProcessorUtil.isChecksumSame(API_CALL, params.checksum, request.getQueryString())) {
+            errors.checksumError()
+            respondWithErrors(errors)
+            return
+        }
 
-    // Do we have a checksum? If none, complain.
-    if (StringUtils.isEmpty(params.checksum)) {
-      errors.missingParamError("checksum");
-    }
+        List<String> recordIdList = new ArrayList<String>();
+        if (!StringUtils.isEmpty(recordId)) {
+            recordIdList = paramsProcessorUtil.decodeIds(recordId);
+        }
 
-    // Do we have a recording id? If none, complain.
-    String recordId = params.recordID
-    if (StringUtils.isEmpty(recordId)) {
-      errors.missingParamError("recordID");
-    }
-    // Do we have a publish status? If none, complain.
-    String publish = params.publish
-    if (StringUtils.isEmpty(publish)) {
-      errors.missingParamError("publish");
-    }
+        if (!meetingService.existsAnyRecording(recordIdList)) {
+            // BEGIN - backward compatibility
+            invalid("notFound", "We could not find recordings");
+            return;
+            // END - backward compatibility
+        }
 
-    if (errors.hasErrors()) {
-      respondWithErrors(errors)
-      return
+        //Execute code specific for this call
+        Map<String, String> metaParams = ParamsProcessorUtil.processMetaParam(params)
+        if (!metaParams.empty) {
+            //Proceed with the update
+            meetingService.updateRecordings(recordIdList, metaParams);
+        }
+        withFormat {
+            xml {
+                render(contentType: "text/xml") {
+                    response() {
+                        returncode(RESP_CODE_SUCCESS)
+                        updated(true)
+                    }
+                }
+            }
+        }
     }
 
-    // Do we agree on the checksum? If not, complain.
-    if (! paramsProcessorUtil.isChecksumSame(API_CALL, params.checksum, request.getQueryString())) {
-      errors.checksumError()
-      respondWithErrors(errors)
-      return
-    }
+    def uploadDocuments(conf) {
+        log.debug("modify on 2018 09 10 ApiController#uploadDocuments(${conf.getInternalId()})");
 
-    ArrayList<String> recordIdList = new ArrayList<String>();
-    if (!StringUtils.isEmpty(recordId)) {
-      recordIdList=paramsProcessorUtil.decodeIds(recordId);
+        String requestBody = request.inputStream == null ? null : request.inputStream.text;
+        requestBody = StringUtils.isEmpty(requestBody) ? null : requestBody;
+        String preName = "中软大讲堂.pdf";
```

Comment:
This is not a general default file name

Reply:
It's OK ,we can change to "default.pdf"

Reply:
but when a few documents be uploaded the system already die.so maybe you can change the code,make them batter.